### PR TITLE
Minor docs updates and cross-references

### PR DESCRIPTION
**Pull request by Zac-HD**
Replaces #722.  And includes some other tweaks while I was there 😉

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -55,8 +50,12 @@
 copyright = u'2013-%s, David R. MacIver' % datetime.datetime.utcnow().year
 author = u'David R. MacIver'
 
-version = __version__
-release = __version__
+_d = {}
+with open(os.path.join(os.path.dirname(__file__), '..', 'src',
+                       'hypothesis', 'version.py')) as f:
+    exec(f.read(), _d)
```
Comment (alexwlchan) [inline docs/conf.py]: Mmm. We already have one copy of this code in `setup.py`, do we need to duplicate it?

File: docs/packaging.rst [hunk 2]
```diff
@@ -98,4 +98,4 @@ Examples
 
 * `arch linux <https://www.archlinux.org/packages/community/any/python-hypothesis/>`_
 * `fedora <https://admin.fedoraproject.org/pkgdb/package/rpms/python-hypothesis/>`_
-* `gentoo <https://packages.gentoo.org/packages/dev-python/hypothesis>`_ (slightly behind at the time of this writing)
+* `gentoo <https://packages.gentoo.org/packages/dev-python/hypothesis>`_
```
Comment (alexwlchan) [inline docs/packaging.rst]: Looks like it’s become very behind now, or have I missed something?

File: docs/conf.py (hunk 1 above)
Comment (Zac-HD) [reply docs/conf.py]: For only two uses in independent scripts, I think it's cleaner to duplicate it than to share.

File: docs/packaging.rst (hunk 2 above)
Comment (Zac-HD) [reply docs/packaging.rst]: The stable (green) version is Hypothesis 3.6 which is pretty old, but they did update the testing (yellow) version to 3.28.2 yesterday.  Either way, it's not slightly behind and that kind of note goes out of date really quickly anyway.

File: docs/packaging.rst [hunk 3]
```diff
@@ -7,8 +7,8 @@ Downstream packagers often want to package Hypothesis. Here are some guidelines.
 The primary guideline is this: If you are not prepared to keep up with the Hypothesis release schedule,
 don't. You will annoy me and are doing your users a disservice.
 
-Hypothesis has quite a frequent release schedule. It's very rare that it goes a month without a release,
-and there are often multiple releases in a given month.
+Hypothesis has a very frequent release schedule. It's rare that it goes a week without a release,
+and there are often multiple releases in a given week.
 
 Many people not only fail to follow the release schedule but also seem included to package versions
```
Comment (mulkieran) [inline docs/packaging.rst]: While you are here... "included" was probably supposed to be "inclined". But you might want to just get rid of the paragraph and one below altogether. I think some of the appearance of packaging stuff that is months out of date is really auto- activities, or global fixes for packaging problems by release engineering or what not.

File: docs/packaging.rst [hunk 4]
```diff
@@ -98,4 +98,4 @@ Examples
 
 * `arch linux <https://www.archlinux.org/packages/community/any/python-hypothesis/>`_
 * `fedora <https://admin.fedoraproject.org/pkgdb/package/rpms/python-hypothesis/>`_
```
Comment (mulkieran) [inline docs/packaging.rst]: While you're here, could you update Fedora link? It should  now be: https://src.fedoraproject.org/rpms/python-hypothesis.

**Comment by Zac-HD** [comment]
OK; I think this is ready to be merged.

File: docs/conf.py (hunk 1 above)
Comment (DRMacIver) [reply docs/conf.py]: It actually appears in a third place FWIW, which is in hypothesistooling.py

Comment (Zac-HD) [reply docs/conf.py]: The basic problem though is that if we could import the snippet we could just import the `__version__` instead, and #722 shows some of the many problems that can cause.

I still think this is the least contorted way of doing the right thing, but would be delighted to take concrete suggestions!

Comment (DRMacIver) [reply docs/conf.py]: I'm OK to leave it as is.

File: docs/conf.py [hunk 5]
```diff
@@ -79,11 +78,8 @@
 # Some standard imports
 from hypothesis import *
 from hypothesis.strategies import *
-# Ensure that output (including from strategies) is deterministic
-import random
-random.seed(0)
-# don't save examples
-settings.register_profile('doctests', settings(database=None))
+# Run deterministically, and don't save examples
+settings.register_profile('doctests', settings(database=None, derandomize=1))
```
[the remark this answers is not in the document]
Comment (DRMacIver) [reply docs/conf.py]: See also #891

File: docs/usage.rst [hunk 6]
```diff
@@ -11,7 +11,7 @@ then it should be available on pypi.
 * `bidict <https://github.com/jab/bidict>`_
 * `binaryornot <https://github.com/audreyr/binaryornot>`_
 * `brotlipy <https://github.com/python-hyper/brotlipy>`_
-* `chardet <https://pypi.python.org/pypi/chardet>`_
+* :pypi:`chardet`
```
Comment (DRMacIver) [inline docs/usage.rst]: I wonder if we should add a similar helper for github (not a requested change, just a thought)

File: docs/conf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Zac-HD) [reply docs/conf.py]: Purely that `1` makes it a one-liner... but I'll change that to `True`.

File: docs/usage.rst (hunk 6 above)
Comment (Zac-HD) [reply docs/usage.rst]: The way custom roles work, the default link text has to be the whole slug, so adding a new bit of custom config doesn't do much:

    `brotlipy <https://github.com/python-hyper/brotlipy>`_
    :github:`brotlipy <python-hyper/brotlipy>`

and it hardly seemed worth the trouble.

**Pull request merged**